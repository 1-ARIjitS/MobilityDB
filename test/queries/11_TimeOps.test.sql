-------------------------------------------------------------------------------
-- Tests of operators that do not involved indexes for time types.
-- File TimeOps.c
-------------------------------------------------------------------------------

SELECT timestamptz '2000-01-01' -|- period '[2000-01-01, 2000-01-03]';
SELECT timestamptz '2000-01-01' -|- period '(2000-01-01, 2000-01-03]';

SELECT timestamptz '2000-01-01' -|- periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';
SELECT timestamptz '2000-01-01' -|- periodset '{(2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';

SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' -|- period '[2000-01-01, 2000-01-03]';
SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' -|- periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';

SELECT period '[2000-01-01, 2000-01-03]' -|- timestamptz '2000-01-01';
SELECT period '[2000-01-01, 2000-01-03]' -|- timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT period '[2000-01-01, 2000-01-03]' -|- period '[2000-01-01, 2000-01-03]';
SELECT period '[2000-01-01, 2000-01-03]' -|- periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';

SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' -|- timestamptz '2000-01-01';
SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' -|- timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' -|- period '[2000-01-01, 2000-01-03]';
SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' -|- periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';

-------------------------------------------------------------------------------

SELECT timestamptz '2000-01-01' + timestamptz '2000-01-01';
SELECT timestamptz '2000-01-01' + timestamptz '2000-01-02';
SELECT timestamptz '2000-01-01' + timestampset '{2000-01-02, 2000-01-03, 2000-01-05}';
SELECT timestamptz '2000-01-01' + timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT timestamptz '2000-01-05' + timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT timestamptz '2000-01-06' + timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT timestamptz '2000-01-01' + period '[2000-01-02, 2000-01-03]';
SELECT timestamptz '2000-01-01' + period '[2000-01-01, 2000-01-03]';
SELECT timestamptz '2000-01-01' + period '(2000-01-01, 2000-01-03]';
SELECT timestamptz '2000-01-02' + period '[2000-01-01, 2000-01-03]';
SELECT timestamptz '2000-01-03' + period '[2000-01-01, 2000-01-03]';
SELECT timestamptz '2000-01-03' + period '[2000-01-01, 2000-01-03)';
SELECT timestamptz '2000-01-05' + period '[2000-01-01, 2000-01-03]';
SELECT timestamptz '2000-01-01' + periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';
SELECT timestamptz '2000-01-01' + periodset '{[2000-01-02, 2000-01-03],[2000-01-04, 2000-01-05]}';
SELECT timestamptz '2000-01-03' + periodset '{[2000-01-02, 2000-01-03],[2000-01-04, 2000-01-05]}';
SELECT timestamptz '2000-01-04' + periodset '{[2000-01-02, 2000-01-03],[2000-01-04, 2000-01-05]}';
SELECT timestamptz '2000-01-04' + periodset '{[2000-01-02, 2000-01-03],[2000-01-05, 2000-01-05]}';
SELECT timestamptz '2000-01-05' + periodset '{[2000-01-02, 2000-01-03],[2000-01-04, 2000-01-05]}';
SELECT timestamptz '2000-01-06' + periodset '{[2000-01-02, 2000-01-03],[2000-01-04, 2000-01-05]}';

SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' + timestamptz '2000-01-01';
SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' + timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' + timestampset '{2000-01-03, 2000-01-05, 2000-01-07}';
SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' + period '[2000-01-01, 2000-01-03]';
SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' + periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';

SELECT period '[2000-01-01, 2000-01-03]' + timestamptz '2000-01-01';
SELECT period '[2000-01-01, 2000-01-03]' + timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT period '(2000-01-01, 2000-01-03]' + timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT period '[2000-01-01, 2000-01-03)' + timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT period '[2000-01-01, 2000-01-03]' + period '[2000-01-01, 2000-01-03]';
SELECT period '[2000-01-01, 2000-01-03]' + period '(2000-01-03, 2000-01-05]';
SELECT period '[2000-01-01, 2000-01-03]' + periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';

SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' + timestamptz '2000-01-01';
SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' + timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' + period '[2000-01-01, 2000-01-03]';
SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' + periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';

-------------------------------------------------------------------------------

SELECT timestamptz '2000-01-01' - timestamp '2000-01-01';
SELECT timestamptz '2000-01-01' - timestamp '2000-01-02';
SELECT timestamptz '2000-01-01' - timestampset '{2000-01-02, 2000-01-03, 2000-01-05}';
SELECT timestamptz '2000-01-01' - timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT timestamptz '2000-01-05' - timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT timestamptz '2000-01-06' - timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT timestamptz '2000-01-01' - period '[2000-01-02, 2000-01-03]';
SELECT timestamptz '2000-01-01' - period '[2000-01-01, 2000-01-03]';
SELECT timestamptz '2000-01-01' - period '(2000-01-01, 2000-01-03]';
SELECT timestamptz '2000-01-02' - period '[2000-01-01, 2000-01-03]';
SELECT timestamptz '2000-01-03' - period '[2000-01-01, 2000-01-03]';
SELECT timestamptz '2000-01-03' - period '[2000-01-01, 2000-01-03)';
SELECT timestamptz '2000-01-05' - period '[2000-01-01, 2000-01-03]';
SELECT timestamptz '2000-01-01' - periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';
SELECT timestamptz '2000-01-01' - periodset '{[2000-01-02, 2000-01-03],[2000-01-04, 2000-01-05]}';
SELECT timestamptz '2000-01-03' - periodset '{[2000-01-02, 2000-01-03],[2000-01-04, 2000-01-05]}';
SELECT timestamptz '2000-01-04' - periodset '{[2000-01-02, 2000-01-03],[2000-01-04, 2000-01-05]}';
SELECT timestamptz '2000-01-04' - periodset '{[2000-01-02, 2000-01-03],[2000-01-05, 2000-01-05]}';
SELECT timestamptz '2000-01-05' - periodset '{[2000-01-02, 2000-01-03],[2000-01-04, 2000-01-05]}';
SELECT timestamptz '2000-01-06' - periodset '{[2000-01-02, 2000-01-03],[2000-01-04, 2000-01-05]}';

SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' - timestamptz '2000-01-01';
SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' - timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' - timestampset '{2000-01-03, 2000-01-05, 2000-01-07}';
SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' - period '[2000-01-01, 2000-01-03]';
SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' - periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';

SELECT period '[2000-01-01, 2000-01-03]' - timestamptz '2000-01-01';
SELECT period '[2000-01-01, 2000-01-03]' - timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT period '(2000-01-01, 2000-01-03]' - timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT period '[2000-01-01, 2000-01-03)' - timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT period '[2000-01-01, 2000-01-03]' - period '[2000-01-01, 2000-01-03]';
SELECT period '[2000-01-01, 2000-01-03]' - period '(2000-01-03, 2000-01-05]';
SELECT period '[2000-01-01, 2000-01-03]' - periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';

SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' - timestamptz '2000-01-01';
SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' - timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' - period '[2000-01-01, 2000-01-03]';
SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' - periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';

-------------------------------------------------------------------------------

SELECT timestamptz '2000-01-01' * timestamp '2000-01-01';
SELECT timestamptz '2000-01-01' * timestamp '2000-01-02';
SELECT timestamptz '2000-01-01' * timestampset '{2000-01-02, 2000-01-03, 2000-01-05}';
SELECT timestamptz '2000-01-01' * timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT timestamptz '2000-01-05' * timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT timestamptz '2000-01-06' * timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT timestamptz '2000-01-01' * period '[2000-01-02, 2000-01-03]';
SELECT timestamptz '2000-01-01' * period '[2000-01-01, 2000-01-03]';
SELECT timestamptz '2000-01-01' * period '(2000-01-01, 2000-01-03]';
SELECT timestamptz '2000-01-02' * period '[2000-01-01, 2000-01-03]';
SELECT timestamptz '2000-01-03' * period '[2000-01-01, 2000-01-03]';
SELECT timestamptz '2000-01-03' * period '[2000-01-01, 2000-01-03)';
SELECT timestamptz '2000-01-05' * period '[2000-01-01, 2000-01-03]';
SELECT timestamptz '2000-01-01' * periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';
SELECT timestamptz '2000-01-01' * periodset '{[2000-01-02, 2000-01-03],[2000-01-04, 2000-01-05]}';
SELECT timestamptz '2000-01-03' * periodset '{[2000-01-02, 2000-01-03],[2000-01-04, 2000-01-05]}';
SELECT timestamptz '2000-01-04' * periodset '{[2000-01-02, 2000-01-03],[2000-01-04, 2000-01-05]}';
SELECT timestamptz '2000-01-04' * periodset '{[2000-01-02, 2000-01-03],[2000-01-05, 2000-01-05]}';
SELECT timestamptz '2000-01-05' * periodset '{[2000-01-02, 2000-01-03],[2000-01-04, 2000-01-05]}';
SELECT timestamptz '2000-01-06' * periodset '{[2000-01-02, 2000-01-03],[2000-01-04, 2000-01-05]}';

SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' * timestamptz '2000-01-01';
SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' * timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' * timestampset '{2000-01-03, 2000-01-05, 2000-01-07}';
SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' * period '[2000-01-01, 2000-01-03]';
SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' * periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';

SELECT period '[2000-01-01, 2000-01-03]' * timestamptz '2000-01-01';
SELECT period '[2000-01-01, 2000-01-03]' * timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT period '(2000-01-01, 2000-01-03]' * timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT period '[2000-01-01, 2000-01-03)' * timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT period '[2000-01-01, 2000-01-03]' * period '[2000-01-01, 2000-01-03]';
SELECT period '[2000-01-01, 2000-01-03]' * period '(2000-01-03, 2000-01-05]';
SELECT period '[2000-01-01, 2000-01-03]' * periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';

SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' * timestamptz '2000-01-01';
SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' * timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' * period '[2000-01-01, 2000-01-03]';
SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' * periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';

-------------------------------------------------------------------------------
