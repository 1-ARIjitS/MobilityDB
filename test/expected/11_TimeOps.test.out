SELECT timestamptz '2000-01-01' -|- period '[2000-01-01, 2000-01-03]';
 ?column? 
----------
 f
(1 row)

SELECT timestamptz '2000-01-01' -|- period '(2000-01-01, 2000-01-03]';
 ?column? 
----------
 t
(1 row)

SELECT timestamptz '2000-01-01' -|- periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';
 ?column? 
----------
 f
(1 row)

SELECT timestamptz '2000-01-01' -|- periodset '{(2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';
 ?column? 
----------
 t
(1 row)

SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' -|- period '[2000-01-01, 2000-01-03]';
 ?column? 
----------
 f
(1 row)

SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' -|- periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';
 ?column? 
----------
 f
(1 row)

SELECT period '[2000-01-01, 2000-01-03]' -|- timestamptz '2000-01-01';
 ?column? 
----------
 f
(1 row)

SELECT period '[2000-01-01, 2000-01-03]' -|- timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
 ?column? 
----------
 f
(1 row)

SELECT period '[2000-01-01, 2000-01-03]' -|- period '[2000-01-01, 2000-01-03]';
 ?column? 
----------
 f
(1 row)

SELECT period '[2000-01-01, 2000-01-03]' -|- periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';
 ?column? 
----------
 f
(1 row)

SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' -|- timestamptz '2000-01-01';
 ?column? 
----------
 f
(1 row)

SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' -|- timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
 ?column? 
----------
 f
(1 row)

SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' -|- period '[2000-01-01, 2000-01-03]';
 ?column? 
----------
 f
(1 row)

SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' -|- periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';
 ?column? 
----------
 f
(1 row)

SELECT period '[2000-01-01, 2000-01-03]' + period '[2000-01-01, 2000-01-03]';
                      ?column?                      
----------------------------------------------------
 {[2000-01-01 00:00:00+00, 2000-01-03 00:00:00+00]}
(1 row)

SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' + timestamptz '2000-01-01';
                                 ?column?                                 
--------------------------------------------------------------------------
 {2000-01-01 00:00:00+00, 2000-01-03 00:00:00+00, 2000-01-05 00:00:00+00}
(1 row)

SELECT timestamptz '2000-01-01' + timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
                                 ?column?                                 
--------------------------------------------------------------------------
 {2000-01-01 00:00:00+00, 2000-01-03 00:00:00+00, 2000-01-05 00:00:00+00}
(1 row)

SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' + timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
                                 ?column?                                 
--------------------------------------------------------------------------
 {2000-01-01 00:00:00+00, 2000-01-03 00:00:00+00, 2000-01-05 00:00:00+00}
(1 row)

SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' + period '[2000-01-01, 2000-01-03]';
                                               ?column?                                               
------------------------------------------------------------------------------------------------------
 {[2000-01-01 00:00:00+00, 2000-01-03 00:00:00+00], [2000-01-04 00:00:00+00, 2000-01-05 00:00:00+00]}
(1 row)

SELECT period '[2000-01-01, 2000-01-03]' + periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';
                                               ?column?                                               
------------------------------------------------------------------------------------------------------
 {[2000-01-01 00:00:00+00, 2000-01-03 00:00:00+00], [2000-01-04 00:00:00+00, 2000-01-05 00:00:00+00]}
(1 row)

SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' + periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';
                      ?column?                      
----------------------------------------------------
 {[2000-01-01 00:00:00+00, 2000-01-05 00:00:00+00]}
(1 row)

SELECT period '[2000-01-01, 2000-01-03]' - period '[2000-01-01, 2000-01-03]';
 ?column? 
----------
 
(1 row)

SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' - timestamptz '2000-01-01';
                     ?column?                     
--------------------------------------------------
 {2000-01-03 00:00:00+00, 2000-01-05 00:00:00+00}
(1 row)

SELECT timestamptz '2000-01-01' - timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
        ?column?        
------------------------
 2000-01-01 00:00:00+00
(1 row)

SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' - timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
         ?column?         
--------------------------
 {2000-01-03 00:00:00+00}
(1 row)

SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' - period '[2000-01-01, 2000-01-03]';
                      ?column?                      
----------------------------------------------------
 {[2000-01-04 00:00:00+00, 2000-01-05 00:00:00+00]}
(1 row)

SELECT period '[2000-01-01, 2000-01-03]' - periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';
 ?column? 
----------
 
(1 row)

SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' - periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';
 ?column? 
----------
 
(1 row)

SELECT period '[2000-01-01, 2000-01-03]' * period '[2000-01-01, 2000-01-03]';
                     ?column?                     
--------------------------------------------------
 [2000-01-01 00:00:00+00, 2000-01-03 00:00:00+00]
(1 row)

SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' * timestamptz '2000-01-01';
 ?column? 
----------
 
(1 row)

SELECT timestamptz '2000-01-01' * timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
 ?column? 
----------
 
(1 row)

SELECT timestampset '{2000-01-01, 2000-01-03, 2000-01-05}' * timestampset '{2000-01-01, 2000-01-03, 2000-01-05}';
                                 ?column?                                 
--------------------------------------------------------------------------
 {2000-01-01 00:00:00+00, 2000-01-03 00:00:00+00, 2000-01-05 00:00:00+00}
(1 row)

SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' * period '[2000-01-01, 2000-01-03]';
                      ?column?                      
----------------------------------------------------
 {[2000-01-01 00:00:00+00, 2000-01-03 00:00:00+00]}
(1 row)

SELECT period '[2000-01-01, 2000-01-03]' * periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';
                      ?column?                      
----------------------------------------------------
 {[2000-01-01 00:00:00+00, 2000-01-03 00:00:00+00]}
(1 row)

SELECT periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}' * periodset '{[2000-01-01, 2000-01-03],[2000-01-04, 2000-01-05]}';
                                               ?column?                                               
------------------------------------------------------------------------------------------------------
 {[2000-01-01 00:00:00+00, 2000-01-03 00:00:00+00], [2000-01-04 00:00:00+00, 2000-01-05 00:00:00+00]}
(1 row)

